Source/Engine/Physics/Joints/Joint.h

```diff
@@ -201,5 +319,57 @@ API_CLASS(Abstract) class FLAXENGINE_API Joint : public Actor
 #endif
     void OnActiveInTreeChanged() override;
     void OnParentChanged() override;
+    void OnParentChangedInHierarchy() override;
     void OnTransformChanged() override;
+
+public:
+#if (FLAXENGINE_VERSION_MAJOR != 2 && FLAXENGINE_VERSION_MINOR >= 0)
+
+    /// <summary>
+    /// [Deprecated in v1.9, removed in v2.0]
+    /// </summary>
+    API_PROPERTY() DEPRECATED Vector3 GetTargetAnchor() const
+    {
+        return LocalPoseActor1.Translation;
+    }
+
+    /// <summary>
+    /// [Deprecated in v1.9, removed in v2.0]
+    /// </summary>
+    API_PROPERTY() DEPRECATED void SetTargetAnchor(const Vector3& value);
+
+    /// <summary>
+    /// [Deprecated in v1.9, removed in v2.0]
+    /// </summary>
+    API_PROPERTY() DEPRECATED Quaternion GetTargetAnchorRotation() const
+    {
+        return LocalPoseActor1.Orientation;
+    }
+
+    /// <summary>
+    /// [Deprecated in v1.9, removed in v2.0]
+    /// </summary>
+    API_PROPERTY() DEPRECATED void SetTargetAnchorRotation(const Quaternion& value);
+
+    /// <summary>
+    /// [Deprecated in v1.9, removed in v2.0]
+    /// </summary>
+    API_FIELD() DEPRECATED ScriptingObjectReference<Actor> Target;
+
+    /// <summary>
+    /// [Deprecated in v1.9, removed in v2.0]
+    /// </summary>
+    API_FUNCTION() DEPRECATED void SetJointLocation(const Vector3& location);
+
+    /// <summary>
+    /// [Deprecated in v1.9, removed in v2.0]
+    /// </summary>
+    API_FUNCTION() DEPRECATED void SetJointOrientation(const Quaternion& orientation);
+#else
+#ifndef STRING2
+#define STRING2(x) #x
+#define STRING(x) STRING2(x)
+#endif
+#pragma message ( __FILE__ "(" STRING(__LINE__) "):" "[Code Mantening] Remove Deprecated code")
+#endif
```

Comment:
Is this needed?

Reply:
yes there is a code inside the engine which is dedicated and expired 
this is reminder